lib/STM_Domain.ml

```diff
@@ -0,0 +1,36 @@
+open STM_Core
+
+module Make (Spec: Spec) = struct
+
+  open Util
+  open QCheck
+  open STM_Core.Make(Spec)
+
+  (* Parallel agreement property based on [Domain] *)
+  let agree_prop_par (seq_pref,cmds1,cmds2) =
+    assume (all_interleavings_ok seq_pref cmds1 cmds2 Spec.init_state);
+    let sut = Spec.init_sut () in
+    let pref_obs = interp_sut_res sut seq_pref in
```

Comment:
This seems to revert #45?

Reply:
Oh, sorry - #45 was only done for `Lin`!
I suggest we do something similar for `STM` - but that it probably better done in a separate PR